circleguard/investigator.py

```diff
@@ -206,11 +206,153 @@ def median_frametime(replay):
 
         Notes
         -----
-        Median is used instead of mean to lessen the effect of outliers.
+        Frametimes on keypresses are added to the next frametime to make frametimes more accurate.
+        Frametimes are further adjusted for random low frametimes that appear.
         """
         # replay.t is cumsum so convert it back to "time since previous frame"
         t = np.diff(replay.t)
-        return np.median(t)
+
+        # filters out insignificant frametimes
+        unique, counts = np.unique(t, return_counts=True)
+        unique = unique[counts > 4]
+        counts = counts[counts > 4]
+
+        # placeholder 0, so t matches k
+        t = np.concatenate(([0], t))
+        replay.k = replay.k[np.isin(t, unique)]
+        t = t[np.isin(t, unique)]
```

Comment:
this is filtering out frames where the frametime appears less than 4 times in the replay? Would prefer to do this by removing outliers instead, using a mathematical definition (eg more than 2 stddevs away from the mean)